org.adempiere.base/src/org/compiere/model/MZoomCondition.java

```diff
@@ -277,6 +277,29 @@ public static int findZoomWindowByTableId(int AD_Table_ID, int recordID)
 	 * @return AD_Window_ID
 	 */
 	public static int findZoomWindowByTableId(int AD_Table_ID, int recordID, int windowNo)
+	{
+		return findZoomWindowByTableIdOrUU(AD_Table_ID, recordID, null, windowNo);
+	}
+
+	/**
+	 * find AD_Window_ID from matching zoom condition record
+	 * @param AD_Table_ID
+	 * @param recordUU
+	 * @return AD_Window_ID
+	 */
+	public static int findZoomWindowByTableUU(int AD_Table_ID, String recordUU, int windowNo)
+	{
+		return findZoomWindowByTableIdOrUU(AD_Table_ID, -1, recordUU, windowNo);
+	}
+
+	/**
+	 * find AD_Window_ID from matching zoom condition record
+	 * @param AD_Table_ID
+	 * @param recordID
+	 * @param recordUU
+	 * @return AD_Window_ID
+	 */
+	public static int findZoomWindowByTableIdOrUU(int AD_Table_ID, int recordID, String recordUU, int windowNo)
```

Comment:
I think this should be private instead of public. Open to misuse since many tables have both ID and UU value.

Reply:
> I think this should be private instead of public. Open to misuse since many tables have both ID and UU value.

Not possible, need to be called from Env class